tmt/steps/discover/shell.py

```diff
@@ -287,6 +287,25 @@ def go(self) -> None:
         assert self.workdir is not None
         testdir = os.path.join(self.workdir, "tests")
 
+        # for imported plans print details about a remote plan
+        parent = self.parent
+        if parent and parent.plan._original_plan:
+            remote_plan_id = self.parent.plan._original_plan._remote_plan_fmf_id
```

Comment:
As expected, mypy does complain about few bits:

> tmt/steps/discover/shell.py:292: error: "Common" has no attribute "plan"  [attr-defined]

I believe this is because of a missing `cast()`. Mypy cannot know, but we do, this could help: `parent = cast(Optional[tmt.steps.discover.Discover], self.parent)`

> tmt/steps/discover/shell.py:293: error: Item "Common" of "Optional[Common]" has no attribute "plan"  [union-attr]
> tmt/steps/discover/shell.py:293: error: Item "None" of "Optional[Common]" has no attribute "plan"  [union-attr]

Mea culpa, this line shouldn't use `self.parent` but `parent` established & tested above, so `remote_plan_id = parent.plan._original_plan._remote_plan_fmf_id` should be enough.